apps/entity-service/src/bounding-boxes/bounding-box.service.ts

```diff
@@ -0,0 +1,271 @@
+      const geojson = JSON.parse(envelope.get("envelope") as string);
+      const coordinates = geojson.coordinates[0];
+
+      for (const point of coordinates) {
+        const [lng, lat] = point;
+        maxLng = Math.max(maxLng, lng);
+        minLng = Math.min(minLng, lng);
+        maxLat = Math.max(maxLat, lat);
+        minLat = Math.min(minLat, lat);
+      }
+    }
+
+    return { minLng, minLat, maxLng, maxLat };
+  }
+
+  private async getBoundingBoxFromGeometries<T extends Model>(
+    model: ModelStatic<T>,
+    whereCondition: WhereOptions,
+    geometryColumn: string,
+    entityType: EntityType,
+    identifier?: string
+  ): Promise<BoundingBoxDto> {
+    const envelopes = await model.findAll({
+      where: whereCondition,
+      attributes: [
+        [Sequelize.fn("ST_ASGEOJSON", Sequelize.fn("ST_Envelope", Sequelize.col(geometryColumn))), "envelope"]
+      ]
+    });
+
+    if (envelopes.length === 0) {
+      const errorMsg =
+        identifier !== undefined && identifier !== null && identifier !== ""
+          ? `No ${entityType.toLowerCase()} found with UUID ${identifier}`
+          : `No ${entityType.toLowerCase()} found with the provided criteria`;
+      throw new NotFoundException(errorMsg);
+    }
+
+    const { minLng, minLat, maxLng, maxLat } = this.extractBoundingBoxFromEnvelopes(envelopes);
+    return this.createBoundingBoxDto(minLng, minLat, maxLng, maxLat);
+  }
+
+  async getPolygonBoundingBox(polygonUuid: string): Promise<BoundingBoxDto> {
+    return this.getBoundingBoxFromGeometries(
+      PolygonGeometry,
+      { uuid: polygonUuid },
+      "geom",
+      EntityType.POLYGON,
+      polygonUuid
+    );
+  }
+
+  async getSiteBoundingBox(siteUuid: string): Promise<BoundingBoxDto> {
+    const site = await Site.findOne({
+      where: { uuid: siteUuid },
+      attributes: ["uuid"]
+    });
+
+    if (site === null || site === undefined) {
+      throw new NotFoundException(`${EntityType.SITE} with UUID ${siteUuid} not found`);
+    }
+
+    const sitePolygons = await SitePolygon.findAll({
+      where: {
+        siteUuid,
+        polygonUuid: { [Op.ne]: "" },
+        isActive: true,
+        deletedAt: null
+      },
+      attributes: ["polygonUuid"]
+    });
+
+    if (sitePolygons.length === 0) {
+      throw new NotFoundException(`No polygons found for site with UUID ${siteUuid}`);
+    }
+
+    const polygonUuids = sitePolygons.map(sp => sp.polygonUuid);
+
+    return this.getBoundingBoxFromGeometries(
+      PolygonGeometry,
+      { uuid: { [Op.in]: polygonUuids } },
+      "geom",
+      EntityType.POLYGON
+    );
+  }
+
+  async getProjectBoundingBox(projectUuid: string): Promise<BoundingBoxDto> {
+    const project = await Project.findOne({
+      where: { uuid: projectUuid },
+      attributes: ["id"]
+    });
+
+    if (project === null || project === undefined) {
+      throw new NotFoundException(`${EntityType.PROJECT} with UUID ${projectUuid} not found`);
+    }
+
```

Comment:
As mentioned in the controller, there should be policy checks on this stuff. In these methods is where I would do it with `await this.policyService.authorize('read', project)` (and in the other entity specific methods in this service)